onnx/defs/quantization/defs.cc

```diff
@@ -101,4 +102,50 @@ ONNX_OPERATOR_SET_SCHEMA(
               updateOutputShape(ctx, 0, input_shape);
             }));
 
+static const char* DynamicQuantizeLinear_ver11_doc = R"DOC(
+A Function to fuse calculation for Scale, Zero Point and FP32->8Bit convertion of FP32 Input data.
+Outputs Scale, ZeroPoint and Quantized Input for a given FP32 Input)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    DynamicQuantizeLinear,
+    11,
+    OpSchema()
+        .SetDoc(DynamicQuantizeLinear_ver11_doc)
+        .Input(0, "x", "Input tensor", "T1")
+        .Output(0, "y", "Quantized output tensor", "T2")
+        .Output(1, "y_scale", "Output scale. It's a scalar, which means a per-tensor/layer quantization.", "tensor(float)")
+        .Output(2, "y_zero_point", "Output zero point. It's a scalar, which means a per-tensor/layer quantization.", "T2")
+        .Attr(
+          "to",
+          "The data type to which the elements of the input tensor are quantized to. Strictly must be one of the types from DataType enum in TensorProto. "
+          "Currently this is required to be uint8 .i.e. value 2.",
```

Comment:
this attribute is actually specifying the value of "T2", so, does that mean only one is needed, otherwise, is there a way to verify to ensure the two have no conflict with each other?

Reply:
The role of this attribute is the same as that in the cast op: it specifies what type to cast to. We need both, since both type-inference and the runtime will choose the final type based on the attribute value. (Of course, it is not needed right now since only one type is supported, but it will be needed if and when multiple output-types are supported.) But you are right about checking/verifying: we could add a shape-and-type-inference function to check the right values are specified … is there some other way to add a simple check?

Reply:
Or, we could omit this for now, since we are not using it currently.

Reply:
No we cannot omit this ... since we need a way to translate T2 to an attribute which Cast can use in the functionBody... plus can we always rely the type info to be present? 

I am in favor of adding a shape inference for this function instead

Reply:
No, we won't translate T2 to the attribute. (Usually, the computation order is the other way: the attribute will be used to infer the value of T2.) However, in this case, you can set the attribute value of Cast to be the constant "2", since that is the only case being supported right now, right?

Reply:
To add more context: I think that once we have the ability to specify the FunctionBody using a lambda, that function will be the right place to do the check: since we may actually generate different bodies based on the value of this attribute. So, it seems like we could postpone this issue to that extension in whatever is the most convenient way.

Reply:
Eventually when we enable this for multiple data types we will need this attribute... Right now if you both think this may cause confusion then I am ok to remove this and make cast attr as a const.

However I expect when a runtime implements kernel for this function it should verify that T2 and attr values dont conflict. This check does not necessarily have to be in ONNX.

Reply:
My suggestion please: if there's an existing way to do the check, it's ok to keep the attribute, otherwise, we should remove it right now and think of adding it as well as a way to do verification.

Reply:
I will remove it right now... let's add it when it is necessary.